modules/heat_conduction/src/materials/SemiconductorLinearConductivity.C

```diff
@@ -0,0 +1,63 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SemiconductorLinearConductivity.h"
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<SemiconductorLinearConductivity>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredCoupledVar("temp", "Variable for temperature in Kelvin.");
+  params.addParam<std::string>("base_name", "Material property base name");
+  params.addRequiredParam<Real>("sh_coeff_A", "Steinhart_Hart coefficient A of the material");
+  params.addRequiredRangeCheckedParam<Real>(
+      "sh_coeff_B", "sh_coeff_B != 0", "Steinhart_Hart coefficient B of the material.");
+  params.addRequiredParam<Real>("T_upper_bound",
+                                "Upper bound of temperature region of applicability");
+  params.addRequiredParam<Real>("T_lower_bound",
+                                "Lower bound of temperature region of applicability");
+  return params;
+}
+
+SemiconductorLinearConductivity::SemiconductorLinearConductivity(const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters),
+    _sh_coeff_A(getParam<Real>("sh_coeff_A")),
+    _sh_coeff_B(getParam<Real>("sh_coeff_B")),
+    _T_upper_bound(getParam<Real>("T_upper_bound")),
+    _T_lower_bound(getParam<Real>("T_lower_bound")),
+    _T(coupledValue("temp")),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _electric_conductivity(declareProperty<Real>(_base_name + "electrical_conductivity")),
+    _delectric_conductivity_dT(declarePropertyDerivative<Real>(
+        _base_name + "electrical_conductivity", getVar("temp", 0)->name()))
+{
+}
+
+void
+SemiconductorLinearConductivity::computeQpProperties()
+{
+  if (MooseUtils::absoluteFuzzyLessThan(_T[_qp], _T_lower_bound))
```

Comment:
Given that we didn't do the divide by zero check because of performance concerns, I'd prefer that we keep the code how you had it before until we can show that this doesn't have a negative performance impact. Also, you could easily end up outside the applicable bounds for the model during nonlinear iterations, but then have the converged solution end up within the bounds, and this would prevent you from ever getting a solution.

Anyway, I'd say that we should remove this bounds check in the initial version of this model, and then discuss options on how to check that it's within bounds in a way that won't affect performance or needlessly generate errors.

Reply:
I changed the temperature bounds in the test, it worked.  I don't know how I got problems with porous_flow.